subsys/pm/device_runtime.c

```diff
@@ -92,6 +92,14 @@ static void runtime_suspend_work(struct k_work *work)
 	k_condvar_broadcast(&pm->condvar);
 	k_mutex_unlock(&pm->lock);
 
+	/*
+	 * On async put, we have to suspend the domain when the device
+	 * finishes its operation
+	 */
+	if (pm->domain != NULL) {
+		(void)pm_device_runtime_put_async(pm->domain);
+	}
```

Comment:
note: this is already called within a work queue context, so why async?

Reply:
This gives the scheduler an opportunity to schedule another thread since this operation does not need to happen synchronously. Though, this is not something  critical, maybe just taking advantage that it is already running in the context of the workqueue and make it synchronously will keep things simpler.